packages/aws-cdk/lib/api/logs/logs-monitor.ts

```diff
@@ -0,0 +1,209 @@
+import * as util from 'util';
+import * as cxapi from '@aws-cdk/cx-api';
+import * as colors from 'colors/safe';
+import { print, error } from '../../logging';
+import { flatten } from '../../util/arrays';
+import { ISDK } from '../aws-auth';
+
+// how often we should read events from CloudWatchLogs
+const SLEEP = 2_000;
+
+/**
+ * Represents a CloudWatch Log Event that will be
+ * printed to the terminal
+ */
+interface CloudWatchLogEvent {
+  /**
+   * The log event message
+   */
+  readonly message: string;
+
+  /**
+   * The name of the log group
+   */
+  readonly logGroupName: string;
+
+  /**
+   * The time at which the event occurred
+   */
+  readonly timestamp: Date;
+}
+
+/**
+ * Configuration tracking information on the log groups that are
+ * being monitored
+ */
+interface LogGroupsAccessSettings {
+  /**
+   * The SDK for a given environment (account/region)
+   */
+  readonly sdk: ISDK;
+
+  /**
+   * A map of log groups and associated startTime in a given account.
+   *
+   * The monitor will read events from the log group starting at the
+   * associated startTime
+   */
+  readonly logGroupsStartTimes: { [logGroupName: string]: number };
+}
+
+export class CloudWatchLogEventMonitor {
+  /**
+   * Determines which events not to display
+   */
+  private startTime: number;
+
+  /**
+   * Map of environment (account:region) to LogGroupsAccessSettings
+   */
+  private readonly envsLogGroupsAccessSettings = new Map<string, LogGroupsAccessSettings>();
+
+  private active = false;
+
+  constructor(startTime?: Date) {
+    this.startTime = startTime?.getTime() ?? Date.now();
+  }
+
+  /**
+   * resume reading/printing events
+   */
+  public activate(): void {
+    this.active = true;
+    this.scheduleNextTick();
```

Comment:
I think this waits 2 seconds before checking for new log group events, and we don't have to wait.

Let's make `scheduleNextTick()` take an argument of how long to wait. Let's pass it as `0` here, and as `SLEEP` in `tick()`.

Reply:
Good idea.